DER/FxDerivatives/FxContracts.rdf

```diff
@@ -75,132 +75,134 @@
 		<fibo-fnd-utl-av:hasMaturityLevel rdf:resource="&fibo-fnd-utl-av;Provisional"/>
 	</owl:Ontology>
 	
-	<owl:Class rdf:about="&fibo-der-fx-ctr;ForeignExchangeDerivative">
+	<owl:Class rdf:about="&fibo-der-fx-ctr;CurrencyDerivative">
 		<rdfs:subClassOf rdf:resource="&fibo-der-rtd-rtd;RateBasedDerivativeInstrument"/>
+		<rdfs:subClassOf rdf:resource="&fibo-fbc-fi-fi;CurrencyInstrument"/>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-der-fx-ctr;hasBuyingCurrency"/>
+				<owl:onClass rdf:resource="&fibo-fnd-acc-cur;Currency"/>
+				<owl:minQualifiedCardinality rdf:datatype="&xsd;nonNegativeInteger">0</owl:minQualifiedCardinality>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-der-fx-ctr;hasSellingCurrency"/>
+				<owl:onClass rdf:resource="&fibo-fnd-acc-cur;Currency"/>
+				<owl:minQualifiedCardinality rdf:datatype="&xsd;nonNegativeInteger">0</owl:minQualifiedCardinality>
+			</owl:Restriction>
+		</rdfs:subClassOf>
 		<rdfs:subClassOf>
 			<owl:Restriction>
 				<owl:onProperty rdf:resource="&fibo-der-drc-bsc;hasUnderlier"/>
 				<owl:someValuesFrom rdf:resource="&fibo-der-rtd-rtd;ForeignExchangeRateObservable"/>
 			</owl:Restriction>
 		</rdfs:subClassOf>
-		<rdfs:label xml:lang="en">foreign exchange derivative</rdfs:label>
+		<rdfs:label xml:lang="en">currency derivative</rdfs:label>
 		<fibo-fnd-utl-av:abbreviation xml:lang="en">FX derivative</fibo-fnd-utl-av:abbreviation>
+		<fibo-fnd-utl-av:synonym xml:lang="en">foreign exchange derivative</fibo-fnd-utl-av:synonym>
 		<fibo-fnd-utl-av:synonym xml:lang="en">forex derivative</fibo-fnd-utl-av:synonym>
 	</owl:Class>
 	
-	<owl:Class rdf:about="&fibo-der-fx-ctr;ForeignExchangeForward">
+	<owl:Class rdf:about="&fibo-der-fx-ctr;CurrencyForward">
 		<rdfs:subClassOf rdf:resource="&fibo-der-drc-ff;Forward"/>
-		<rdfs:subClassOf rdf:resource="&fibo-der-fx-ctr;ForeignExchangeDerivative"/>
-		<rdfs:label xml:lang="en">foreign exchange forward</rdfs:label>
+		<rdfs:subClassOf rdf:resource="&fibo-der-fx-ctr;CurrencyDerivative"/>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-der-fx-ctr;hasForwardExchangeRate"/>
+				<owl:someValuesFrom rdf:resource="&fibo-fnd-acc-cur;ExchangeRate"/>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:label xml:lang="en">currency forward</rdfs:label>
 		<skos:definition xml:lang="en">agreement to deliver and settle a given amount of currency in one currency, in exchange for a given amount in another currency, at an agreed date in the future and at an agreed rate of exchange</skos:definition>
```

Comment:
"amount of currency in one currency" seems wrong. Better "amount of money in one currency"

Reply:
done